spreadsheets.js

```diff
@@ -0,0 +1,215 @@
+var moment = require("moment");
+var fuzzy_match = require("./fuzzy_match");
+var GoogleSpreadsheet = require("google-spreadsheet");
+
+var ISO_TUESDAY = 2;
+var DATE_FORMATS = [
+    "YYYYMMDD HHmm",
+    "YYYYMMDD Hmm",
+    "HHmm DDMMYYYY"
+];
+var WORDS_TO_IGNORE = [
+    "",
+    "-",
+    "–",
+    "at",
+    "on",
+    "gmt",
+    "rescheduled",
+    "reschedule",
+    "reschedule:",
+    "to",
+    "v",
+    "vs",
+    "vs.",
+    "versus",
+    "sunday",
+    "monday",
+    "tuesday",
+    "wednesday",
+    "thursday",
+    "friday",
+    "saturday",
+    "sun",
+    "mon",
+    "tues",
+    "wed",
+    "thu",
+    "thur",
+    "thurs",
+    "fri",
+    "sat",
+
+    // TODO: yes, this is oddly specific - but unless he changes his username ....
+    "steveharwell"
+];
+
+// A scheduling error (as opposed to other errors)
+function SchedulingError () {}
+SchedulingError.prototype = new Error();
+function PairingError () {}
+PairingError.prototype = new Error();
+
+// Make an attempt to parse a scheduling string. 
+//
+// first parameter is the string to parse
+// second parameter is an options dict specifying the round extrema options
+function parse_scheduling(string, options) {
+    // Do some basic preprocessing
+    string = string.replace(/[@\.,\(\)\<\>]/g, ' ');
+
+    var parts = string.split(" ");
+
+    // Filter it to remove all the noise
+    var filtered_parts = [];
+    parts.forEach(function(part) {
+        // Remove all lone 
+        if (WORDS_TO_IGNORE.indexOf(part.toLowerCase()) != -1) {
+            return;
+        }
+        filtered_parts.push(part);
+    });
+    if (filtered_parts.length < 3) {
+        console.log("Unable to parse scheduling string: " + string);
+        throw new SchedulingError();
+    }
+
+    // Now build up some possible strings and try a bunch of patterns
+    // to find a date in our range.
+    var extrema = get_round_extrema(options);
```

Comment:
please confirm my understanding here:

essentially, this will match if its

`[ignorable strings] @<opponent1> [ignorable strings] @<opponent2> [ignorable strings] <datestring> [ignorable strings]`

yeah?